src/common/util/colors.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * The interface used for formatting strings to contain color metadata.
+ *
+ * Use the interface properties to construct a style, then use the
+ * `(s: string): string` function to format the provided string with the given
+ * style.
+ */
+export interface Colors {
```

Comment:
maybe it's possible to reuse types definitions from ansi-colors? not strictly necessary though